src/main/java/net/fabricmc/loom/util/ModProcessor.java

```diff
@@ -58,82 +55,34 @@
 import net.fabricmc.loom.providers.MinecraftMappedProvider;
 import net.fabricmc.loom.util.accesswidener.AccessWidener;
 import net.fabricmc.loom.util.accesswidener.AccessWidenerRemapper;
-import net.fabricmc.tinyremapper.OutputConsumerPath;
+import net.fabricmc.loom.processors.dependency.ModDependencyInfo;
 import net.fabricmc.tinyremapper.TinyRemapper;
+import net.fabricmc.tinyremapper.InputTag;
+import net.fabricmc.tinyremapper.OutputConsumerPath;
 
 public class ModProcessor {
-	private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
-
-	public static void processMod(File input, File output, Project project, Configuration config, ResolvedArtifact artifact) throws IOException {
-		if (output.exists()) {
-			output.delete();
-		}
+	public static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
 
-		remapJar(input, output, project, artifact);
-
-		//Enable this if you want your nested jars to be extracted, this will extract **all** jars
-		if (project.getExtensions().getByType(LoomGradleExtension.class).extractJars) {
-			handleNestedJars(input, project, config, artifact);
+	public static void processMods(Project project, List<ModDependencyInfo> processList) throws IOException {
+		if (processList.isEmpty()) {
+			return;
 		}
 
-		remapaccessWidener(output, project);
-
-		//Always strip the nested jars
-		stripNestedJars(output);
-	}
-
-	private static void handleNestedJars(File input, Project project, Configuration config, ResolvedArtifact artifact) throws IOException {
-		try (JarFile jarFile = new JarFile(input)) {
-			JarEntry modJsonEntry = jarFile.getJarEntry("fabric.mod.json");
-
-			if (modJsonEntry == null) {
-				return;
+		for (ModDependencyInfo info : processList) {
+			if (info.getRemappedOutput().exists()) {
+				info.getRemappedOutput().delete();
```

Comment:
Does this get called every time you refresh gradle or is there something preventing it after the first time?

Reply:
This should never be reached. I should remove it or throw here.